adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala

```diff
@@ -576,32 +594,67 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
     ))
   }
 
-  def loadVcf(filePath: String, sd: Option[SequenceDictionary]): RDD[VariantContext] = {
+  /**
+   * Loads a VCF file into an RDD.
+   *
+   * @param filePath The file to load.
+   * @param sdOpt An optional sequence dictionary, in case the sequence info
+   *              is not included in the VCF.
+   * @return Returns a VariantContextRDD.
+   */
+  def loadVcf(filePath: String,
+              sdOpt: Option[SequenceDictionary] = None): VariantContextRDD = {
+
+    // load vcf data
     val job = HadoopUtil.newJob(sc)
-    val vcc = new VariantContextConverter(sd)
     job.getConfiguration().set("io.compression.codecs", classOf[BGZFCodec].getCanonicalName())
+    val vcc = new VariantContextConverter(sdOpt)
     val records = sc.newAPIHadoopFile(
       filePath,
       classOf[VCFInputFormat], classOf[LongWritable], classOf[VariantContextWritable],
       ContextUtil.getConfiguration(job)
     )
+
+    // attach instrumentation
     if (Metrics.isRecording) records.instrument() else records
 
-    records.flatMap(p => vcc.convert(p._2.get))
+    // load vcf metadata
+    val (vcfSd, samples) = loadVcfMetadata(filePath)
+
+    // we can only replace the sequences header if the sequence info was missing on the vcf
+    require(sdOpt.isEmpty || vcfSd.isEmpty,
+      "Only one of the provided or VCF sequence dictionary can be specified.")
+    val sd = sdOpt.getOrElse(vcfSd)
+
+    VariantContextRDD(records.flatMap(p => vcc.convert(p._2.get)),
+      sd,
+      samples)
   }
 
   def loadParquetGenotypes(
     filePath: String,
     predicate: Option[FilterPredicate] = None,
-    projection: Option[Schema] = None): RDD[Genotype] = {
-    loadParquet[Genotype](filePath, predicate, projection)
+    projection: Option[Schema] = None): GenotypeRDD = {
+    val rdd = loadParquet[Genotype](filePath, predicate, projection)
+
+    // load sequence info
+    val sd = loadAvroSequences(filePath)
+
+    // load avro record group dictionary and convert to samples
+    val rgd = loadAvroSampleMetadata(filePath)
```

Comment:
The issue is with not producing an `_rgdict.avro`. 
I used `loadGenotypes` in the adam shell and it threw an error because the `_rgdict.avro` wasn't there.